spec/models/asset_spec.rb

```diff
@@ -304,4 +310,48 @@
       expect(asset.deleted_at).to be_nil
     end
   end
+
+  describe "extension" do
+    context "when asset file has extension" do
+      let(:asset) { Asset.new(file: load_fixture_file("asset.png")) }
+
+      it "returns asset file extension" do
+        expect(asset.extension).to eq('png')
+      end
+    end
+
+    context "when asset file has capitalised extension" do
+      let(:asset) { Asset.new(file: load_fixture_file("asset-with-capitalised-extension.TXT")) }
+
+      it "returns downcased extension" do
+        expect(asset.extension).to eq('txt')
+      end
+    end
+
+    context "when asset file has no extension" do
+      let(:asset) { Asset.new(file: load_fixture_file("asset-without-extension")) }
+
+      it "returns empty string" do
+        expect(asset.extension).to eq('')
+      end
+    end
+  end
+
+  describe "content_type" do
+    context "when asset file has extension" do
+      let(:asset) { Asset.new(file: load_fixture_file("asset.png")) }
+
+      it "returns content type based on asset file extension" do
+        expect(asset.content_type).to eq(Mime::Type.lookup('image/png').to_s)
+      end
+    end
+
+    context "when asset file has no extension" do
+      let(:asset) { Asset.new(file: load_fixture_file("asset-without-extension")) }
+
+      it "returns default content type" do
+        expect(asset.content_type).to eq(AssetManager.default_content_type)
```

Comment:
I wonder if it might be worth asserting the literal value of `ActionController::DataStreaming::DEFAULT_SEND_FILE_TYPE` here, just in case that constant changes to `nil` or `""` when we upgrade Rails.